V0.1.2

dmcharlie19 opened the pull request:
реализованы BE и FE приложения

dmcharlie19 committed:
backend приложение

Реализован контроллер Recipe, сделаны операции CRUD
Реализован сервис Recipe
Реализован репозиторий Recipe
настроена БД

dmcharlie19 committed:
приложение Frontend

+проект приложения
+ страница для просмотра рецептов
+ Модель рецепта на FE
+Связать FE и BE через HttpClient, метод на BE для получения рецепта

dmcharlie19 committed:
изменена организация BE приложения

файлы распределены по слоям
под каждый слой создан проект

dmcharlie19 committed:
Восстановление собираемости бэка

переименования пространств имен согласно слоям
настройка зависимостей между проектами

egoshin-igor commented on CookBookBackend/CookBook.Application/AppServices/IRecipeService.cs:
```diff
@@ -0,0 +1,14 @@
+{
+  public interface IRecipeService
+  {
+    List<RecipeDto> GetRecipes();
```
Т.к. это RecipeService то Recipes писать не обязательно по контексту понятно
_recipeService.GetRecipes() => _recipeService.GetAll().
CreateRecipe -> Create
GetRecipe -> GetById и т.п.

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookBackend/CookBook.Application/AppServices/RecipeService.cs:
```diff
@@ -0,0 +1,69 @@
+      };
+
+      _recipeRepository.Create( recipe );
+      _unitOfWork.Commit();
```
uow commit на самом высоком уровне. В контроллере. 
Можно в CreateRecipe вернуть recipe, а и потом id вернуть

dmcharlie19 replied:
ок, пока не актуально

egoshin-igor commented on CookBookBackend/CookBook.Api/Controllers/RecipeController.cs:
```diff
@@ -0,0 +1,105 @@
+    {
+      try
+      {
+        return _recipeService.GetRecipes().Select( x => new RecipeCommand
```
По названиям. Наружу все равно возвращаются Dto, т.е. RecipeDto, который лежит в Api так же.
Тут можно так же назвать RecipeDto или чтобы конфликтов в названии не было CroppedRecipeDto - лучше не придумал. Можно просто RecipeDto тоже, но тогда полное имя придется писать с неймспейсом
Для детального рецепта по id -- RecipeDetailDto.
Так же RecipeService можно разделить на две --- чтение и изменение -- RecipeQuery и RecipeManager
Вторая вариант ----
Получать сущности рецепта прям тут в контроллере и превращать в дтошку RecipeDto или RecipeDetailsDto

dmcharlie19 replied:
Решил убрать Dto с уровня апи. Оставить на уровне Апп

egoshin-igor commented on CookBookBackend/CookBook.Api/Controllers/RecipeController.cs:
```diff
@@ -0,0 +1,105 @@
+    {
+      try
+      {
+        return _recipeService.GetRecipes().Select( x => new RecipeCommand
```
Так же сделай папку Mappers. и там класс RecipeMapper
с методами расширения 
для второго варианта
public static RecipeDto Map(this Recipe recipe)
public static RecipeDetails Map(this Recipe recipe)
для первого варианта
public static RecipeDto Map(this RecipeDto recipe)
public static RecipeDetails Map(this RecipeDto recipe)

А тут просто будешь вызывать просто x.Map()

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookBackend/CookBook.Api/Controllers/RecipeController.cs:
```diff
@@ -0,0 +1,105 @@
+    [HttpGet( "get-all" )]
+    public List<RecipeCommand>? GetAll()
+    {
+      try
```
try catch везде можешь убрать. Потом можно просто middleware написать можно который исключения ловит и определенные ошибки кидает

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookBackend/CookBook.Api/Controllers/RecipeController.cs:
```diff
@@ -0,0 +1,105 @@
+      _recipeService = recipeService;
+    }
+
+    [HttpGet( "get-all" )]
```
тут просто пустая строка ---- api/recipes и так понятно что все по рест апи

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookBackend/CookBook.Api/Controllers/RecipeController.cs:
```diff
@@ -0,0 +1,105 @@
+    }
+
+    [HttpPost( "create" )]
+    public IActionResult Create( [FromBody] RecipeCommand recipeCommand )
```
Раздели дтошки. Для создания рецепта и для получения рецепта и для изменения рецепта могут понадобиться разные данные
RecipeDetailsDto
RecipeDto
CreateRecipeCommand
EditRecipeCommandDto

dmcharlie19 replied:
эти методы я пока удалю, пока чтомне не понятно какими они должны быть в итоге

egoshin-igor commented on CookBookBackend/CookBook.Api/Controllers/RecipeController.cs:
```diff
@@ -0,0 +1,105 @@
+    [HttpPost( "create" )]
+    public IActionResult Create( [FromBody] RecipeCommand recipeCommand )
+    {
+      if ( recipeCommand.Title == null )
```
валидацию в сервисе можешь делать и исключения кидать. некорректные ситуации
Потом какой нить свой Exception будешь кидать --- ValidationException например.
А в middleware ловить и возвращать BadRequest

dmcharlie19 replied:
ок, также на потом откладывается

egoshin-igor commented on CookBookBackend/CookBook.Api/Controllers/RecipeController.cs:
```diff
@@ -0,0 +1,105 @@
+
+      try
+      {
+        int id = _recipeService.CreateRecipe( new RecipeDto
```
тут тоже маппер из CreateRecipeCommandDto в CreateRecipeCommand. А в _recipeService.CreateRecipe принимать CreateRecipeCommand

dmcharlie19 replied:
аналогично предыдущему

egoshin-igor commented on CookBookBackend/CookBook.Api/Controllers/RecipeController.cs:
```diff
@@ -0,0 +1,105 @@
+    }
+
+    [HttpPost( "create" )]
+    public IActionResult Create( [FromBody] RecipeCommand recipeCommand )
```
добавление рецепта только в версии v2.0. Не стоит спешить

dmcharlie19 replied:
ок, удалил

egoshin-igor commented on CookBookBackend/CookBook.Api/Controllers/RecipeController.cs:
```diff
@@ -0,0 +1,105 @@
+    {
+      try
+      {
+        _recipeService.Delete( recipeId );
```
DeleteById

dmcharlie19 replied:
удалил пока что сетод

egoshin-igor commented on CookBookBackend/CookBook.Api/Program.cs:
```diff
@@ -0,0 +1,48 @@
+string connection = builder.Configuration.GetConnectionString( "DefaultConnection" );
+builder.Services.AddDbContext<CookBookDbContext>( x => x.UseSqlServer( connection ) );
+
+// Контороллеры
```
опечатка

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookBackend/CookBook.Api/Program.cs:
```diff
@@ -0,0 +1,48 @@
+builder.Services.AddControllers();
+
+//DI
+builder.Services.AddScoped<IRecipeRepository, RecipeRepository>();
```
для Di можно расширения писать
Что то типа ApplicationBindings в слое Application(AddApplication(this IServiceCollection)
InfrastructureBindings и т.п.

dmcharlie19 replied:
не совсем понял что ты предлагаешь

egoshin-igor commented on CookBookBackend/CookBook.Api/Program.cs:
```diff
@@ -0,0 +1,48 @@
+app.MapControllers();
+
+// CORS
+app.UseCors( builder => builder.WithOrigins( "http://localhost:4200" ) );
```
это костыль конечно. А так тебе потом nginx нужно добавить

egoshin-igor replied:
и без корс будет работать

dmcharlie19 replied:
ок, пока оставлю, потом выпилю наверное

egoshin-igor commented on CookBookBackend/CookBook.Api/Properties/launchSettings.json:
```diff
@@ -0,0 +1,29 @@
+      "applicationUrl": "http://localhost:5277",
+      "dotnetRunMessages": true
+    },
+    "IIS Express": {
```
это можешь убрать. Вряд ли иис юзаешь. Консольки достаточно

dmcharlie19 replied:
ок убрал

egoshin-igor commented on CookBookBackend/CookBook.Application/AppServices/RecipeService.cs:
```diff
@@ -0,0 +1,69 @@
+
+    public void Delete( int recipeId )
+    {
+      _recipeRepository.Delete( recipeId );
```
нужно проверка что такой рецепт существует. Если существует то исключение кидать

dmcharlie19 replied:
ок, остается на будущее

egoshin-igor commented on CookBookBackend/CookBook.Application/AppServices/RecipeService.cs:
```diff
@@ -0,0 +1,69 @@
+      _unitOfWork.Commit();
+    }
+
+    public RecipeDto? GetRecipe( int recipeId )
```
тут и в GetRecipes разные дто будут. Для деталей больше данных. Для списка рецептов меньше

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookBackend/CookBook.Infrastructure/Foundation/CookBookDbContext.cs:
```diff
@@ -0,0 +1,22 @@
+{
+  public class CookBookDbContext : DbContext
+  {
+    public DbSet<Recipe> RecipeSet { get; set; }
```
Просто Recipes назвать?

dmcharlie19 replied:
можно

egoshin-igor commented on CookBookBackend/CookBook.Infrastructure/Repositories/RecipeRepository.cs:
```diff
@@ -0,0 +1,49 @@
+
+    public List<Recipe> GetRecipes()
+    {
+      return _dbContext.RecipeSet.ToList();
```
тут потом не будет списка. будет фильтрация, т.к. не все рецепты показываем. Но это потом

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookFrontend/angular.json:
```diff
@@ -0,0 +1,128 @@
+        "serve": {
+          "builder": "@angular-devkit/build-angular:dev-server",
+          "options": {
+            "browserTarget": "TodoList:build"
```
тут и ниже неправильные названия ---TodoList

dmcharlie19 replied:
ок, заменил

egoshin-igor commented on CookBookFrontend/src/app/common-styles.css:
```diff
@@ -0,0 +1,108 @@
+:root{
+    --yellow: #FDB100;
```
это комментарии? Убрать нужно, если не юзаются

egoshin-igor replied:
Или это что то другое. Если другое то отступы поправь

dmcharlie19 replied:
убрал

egoshin-igor commented on CookBookFrontend/src/app/common-styles.css:
```diff
@@ -0,0 +1,108 @@
+	--standartMargin: 5px 0;
+}
+
+.body{
```
форматирование. Поставь какое нить автоформатирование в ui

dmcharlie19 replied:
ок, настроил форматирование по сохранению

egoshin-igor commented on CookBookFrontend/src/app/common-styles.css:
```diff
@@ -0,0 +1,108 @@
+	display: flex;
+	align-items:center;
+
+	/* margin: 0 8px;
```
комментарии лишние

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookFrontend/src/app/common-styles.css:
```diff
@@ -0,0 +1,108 @@
+	margin: 5px 0; /* внешний отступ */
+}
+
+.Navigation{
```
наименование css классов в ститле name-subname-etc. Navigation -> navigation

dmcharlie19 replied:
ок, постарался везде переименовать

egoshin-igor commented on CookBookFrontend/src/app/common-styles.css:
```diff
@@ -0,0 +1,108 @@
+		color: var(--grey);
+	}
+
+	.CurrentLink
```
единое наименование стилей юзай

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookFrontend/src/app/common-styles.css:
```diff
@@ -0,0 +1,108 @@
+		color: #3A3C41;
+	}
+
+.User{
```
отступы

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookFrontend/src/app/common-styles.css:
```diff
@@ -0,0 +1,108 @@
+	padding: inherit; */
+}
+
+	.recipes
```
это точно тут, а не в другом компоненте?

dmcharlie19 replied:
вынес заголовок и футер в отдельные компоненты

egoshin-igor commented on CookBookFrontend/src/app/recipes-list/recipes-list.component.html:
```diff
@@ -0,0 +1,60 @@
+                </div>
+                <div class="MainField">
+                    <p class="RecipeTitle" >{{recipe.title}}</p>
+                    <div style="height: 15px;"></div>
```
стили напрямую не пиши. В классы выводи

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookFrontend/src/app/recipes-list/recipes-list.component.html:
```diff
@@ -0,0 +1,60 @@
+
+        <div style="height: 50px;"></div>
+        <div class="RecipeList" *ngFor="let recipe of recipes;">
+            <div class="OneRecipeBox">
```
карточку рецепты вынеси в отдельный компонент. Он независим и переиспользуемым будет.
recipe-card

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookFrontend/src/app/recipes-list/recipes-list.component.ts:
```diff
@@ -0,0 +1,40 @@
+  }
+    
+    // получаем данные через сервис
+    LoadTodos() {
```
неправильное название метода

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookFrontend/src/app/recipes-list/recipes-list.component.ts:
```diff
@@ -0,0 +1,40 @@
+  }
+    
+    // получаем данные через сервис
+    LoadTodos() {
```
именование методов с маленькой буквы.

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookFrontend/src/app/recipes-list/recipes-list.component.ts:
```diff
@@ -0,0 +1,40 @@
+            }
+          },
+          (error) => {
+            console.log("failed load recipes");
```
тут в ui что то должно показаться

dmcharlie19 replied:
ок, сделал

egoshin-igor commented on CookBookFrontend/src/app/services/recipeService.ts:
```diff
@@ -0,0 +1,20 @@
+@Injectable()
+export class RecipeService {
+
+    private url = "/api/Recipe/";
```
слэш убери в конце. ниже увидишь почему

egoshin-igor commented on CookBookFrontend/src/app/services/recipeService.ts:
```diff
@@ -0,0 +1,20 @@
+    }
+
+    GetRecipes() {
+        return this.http.get(this.url + "get-all").pipe(catchError(err => {  
```
интерполяция есть `${this.url}/get-all`

egoshin-igor commented on CookBookBackend/CookBook.Api/Command/RecipeCommand.cs:
```diff
@@ -0,0 +1,10 @@
+namespace CookBookBackend.Api.Command
```
файлов миграций  не хватает для разворачивания базы?

dmcharlie19 replied:
добавил миграции

egoshin-igor commented on CookBookBackend/CookBook.Api/Command/RecipeCommand.cs:
```diff
@@ -0,0 +1,10 @@
+namespace CookBookBackend.Api.Command
+{
+  public class RecipeCommand
```
Не соответствует дизайну надо поправить
![image](https://user-images.githubusercontent.com/24194554/165389713-41a12f65-d106-42e5-8f7d-f7fe9c40f919.png)

egoshin-igor replied:
Ссылки никак не выделяются

egoshin-igor replied:
Размеры кнопок не такие как на дизайне

egoshin-igor replied:
Можешь md bootstrap подрубить или сам доверстать. в md-bootstrap такие кнопки есть.

egoshin-igor replied:
Короче приведи к дизайну. Сделай чтобы кнопки были такие как на дизайне, отображались как кликабельные. footer чтобы не так сильно примыкал - он внизу страницы должен быть. Опечатки поправь(от шев-поваров)
Шрифты нужные скачай

egoshin-igor replied:
Дизайн важен. Не примут на финальной защите

dmcharlie19 replied:
ок, доработал стиль

egoshin-igor commented on CookBookBackend/CookBook.Api/Controllers/RecipeController.cs:
```diff
@@ -0,0 +1,105 @@
+{
+  [ApiController]
+  [Route( "api/[controller]" )]
+  public class RecipeController : ControllerBase
```
RecipeController -> RecipesController

dmcharlie19 replied:
ок

dmcharlie19 committed:
работа над стилем

Доработка страницы с рецептами
в модель рецепта добавлены новые поля: лайки теги избранное

dmcharlie19 committed:
доработка бэкенда

добалены миграции
В модель рецепта добавлены ноывые поля
Удалены лишние классы (сервис и репозиторий)
Получение рецептов перенесено в класс Query

dmcharlie19 committed:
вынос заголовка и футера в отдельные комплненьы

и небольшие доработки по дизайну

dmcharlie19 committed:
вынос карточки рецепта в отдельный компонент

+добавлены иконки

dmcharlie19 committed:
переименование контроллера

egoshin-igor commented on CookBookBackend/CookBook.Api/Controllers/RecipesController.cs:
```diff
@@ -0,0 +1,25 @@
+    [HttpGet( "" )]
+    public List<RecipeDto>? GetAll()
+    {
+      return _recipeQuery.GetAll();
```
тут правильно будет еще дтошки слоя application менять в дто слоя api. Я уже писал об этом - контракты которые предоставляет слой application может отличаться от контрактов, которые выдает api. Если хочешь, то можешь оставить так, т.к. в рамках проекта вряд ли другие кейсы использования query будут,  но я не рекомендую

dmcharlie19 replied:
да, я понимаю, но специально не стал делать т.к смыла не увидел, по факут будут две одинаковые DTO.

dmcharlie19 replied:
предлагаю пока так оставить. Это же не является ошибкой. Зависимости правильно напрвлены

egoshin-igor commented on CookBookBackend/CookBook.Core/Domain/Recipe.cs:
```diff
@@ -0,0 +1,13 @@
+{
+  public class Recipe
+  {
+    public int Id { get; set; }
```
Давай сделаем сеттеры приватными и изменять сущность в будущем будем через методы, а создавать через конструкторы. Так твоя сущность станет прозрачнее

dmcharlie19 replied:
ок, сделаю

egoshin-igor commented on CookBookBackend/CookBook.Infrastructure/Repositories/UnitOfWorks.cs:
```diff
@@ -0,0 +1,20 @@
+
+namespace CookBookBackend.Infrastructure.Repositories
+{
+  public class UnitOfWork : IUnitOfWork
```
Название файла пофикси

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookFrontend/src/app/common-styles.css:
```diff
@@ -0,0 +1,39 @@
+.onHover-sizeUp {
```
название стилей не везде пофиксил

dmcharlie19 replied:
тут надо исправить на on-hover-size-up?

egoshin-igor commented on CookBookBackend/CookBook.Api/Migrations/20220430140646_AddNewFieldsInRecipe.cs:
```diff
@@ -0,0 +1,117 @@
+
+namespace To_Do_List_Backend.Migrations
+{
+    public partial class AddNewFieldsInRecipe : Migration
```
обычно пишут AlterRecipe_AddNewFields

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookBackend/CookBook.Api/Properties/Resources.resx:
```diff
@@ -0,0 +1,101 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
может выкинуть?

egoshin-igor commented on CookBookBackend/CookBook.Api/Properties/launchSettings.json:
```diff
@@ -0,0 +1,13 @@
+{
+  "$schema": "https://json.schemastore.org/launchsettings.json",
+  "profiles": {
+    "To_Do_List_Backend": {
```
TODOList

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookBackend/CookBook.Application/Mappers/RecipeMapper.cs:
```diff
@@ -0,0 +1,23 @@
+
+namespace CookBook.Application.Mappers
+{
+  public class RecipeMapper
```
public static добавить

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookBackend/CookBook.Application/Mappers/RecipeMapper.cs:
```diff
@@ -0,0 +1,23 @@
+{
+  public class RecipeMapper
+  {
+    public static RecipeDto Map( Recipe r )
```
Map( this Recipe r )

RecipeMapper.Map( r ) -> r.Map()

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookBackend/CookBook.Application/Mappers/RecipeMapper.cs:
```diff
@@ -0,0 +1,23 @@
+{
+  public class RecipeMapper
+  {
+    public static RecipeDto Map( Recipe r )
```
r -> recipe

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookBackend/CookBook.Application/Queries/IRecipeQuery.cs:
```diff
@@ -0,0 +1,14 @@
+{
+  public interface IRecipeQuery
+  {
+    List<RecipeDto> GetAll();
```
IReadOnlyList лучше возвращать

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookFrontend/package.json:
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "todo-list",
```
todo

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookFrontend/src/app/app.component.ts:
```diff
@@ -0,0 +1,9 @@
+
+@Component({
+    selector: 'app-root',
+    template: `<div>
```
в html?

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookFrontend/src/app/common-styles.css:
```diff
@@ -0,0 +1,39 @@
+.onHover-sizeUp {
```
hovered-link {
}

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookFrontend/src/app/recipes-list/recipes-list.component.ts:
```diff
@@ -0,0 +1,59 @@
+
+  // получаем данные через сервис
+  loadRecipes() {
+
```
пустая строка лишняя

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookFrontend/src/app/recipes-list/recipes-list.component.ts:
```diff
@@ -0,0 +1,59 @@
+  // получаем данные через сервис
+  loadRecipes() {
+
+    this.todoService.GetRecipes().subscribe(
```
todo

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookFrontend/src/app/recipes-list/recipes-list.component.ts:
```diff
@@ -0,0 +1,59 @@
+    this.todoService.GetRecipes().subscribe(
+      (data: Recipe[]) => {
+        for (var i = 0; i < data.length; i++) {
+          this.recipes.push(data[i]);
```
можно все за раз запушить

dmcharlie19 replied:
ок

egoshin-igor commented on CookBookFrontend/src/app/recipes-list/recipes-list.component.ts:
```diff
@@ -0,0 +1,59 @@
+      },
+      (error) => {
+        console.log("failed load recipes");
+        this.isServerError = true;
```
isLoaded

dmcharlie19 replied:
ok

dmcharlie19 committed:
незначительные доработки

dmcharlie19 committed:
исправления наименований
